gcsio/src/main/java/com/google/cloud/hadoop/gcsio/authorization/StorageRequestAuthorizer.java

```diff
@@ -0,0 +1,139 @@
+        authorizationHandler.handleInsertObject(
+            createGCSURI(
+                bucketName,
+                ((StorageObject) ((JsonHttpContent) request.getHttpContent()).getData())
+                    .getName()));
+      } else if (request instanceof Storage.Objects.Compose) {
+        URI destinationURI =
+            createGCSURI(bucketName, ((Storage.Objects.Compose) request).getDestinationObject());
+        // read all source objects
+        List<URI> sourceList = new ArrayList<>();
+        for (ComposeRequest.SourceObjects source :
+            ((ComposeRequest) ((JsonHttpContent) request.getHttpContent()).getData())
+                .getSourceObjects()) {
+          sourceList.add(createGCSURI(bucketName, source.getName()));
+        }
+        authorizationHandler.handleComposeObject(destinationURI, sourceList);
+      } else if (request instanceof Storage.Objects.Get) {
+        authorizationHandler.handleGetObject(
+            createGCSURI(bucketName, ((Storage.Objects.Get) request).getObject()));
+      } else if (request instanceof Storage.Objects.Delete) {
+        authorizationHandler.handleDeleteObject(
+            createGCSURI(bucketName, ((Storage.Objects.Delete) request).getObject()));
+      } else if (request instanceof Storage.Objects.Rewrite) {
+        authorizationHandler.handleRewriteObject(
+            createGCSURI(
+                ((Storage.Objects.Rewrite) request).getSourceBucket(),
+                ((Storage.Objects.Rewrite) request).getSourceObject()),
+            createGCSURI(
+                ((Storage.Objects.Rewrite) request).getDestinationBucket(),
+                ((Storage.Objects.Rewrite) request).getDestinationObject()));
+
+      } else if (request instanceof Storage.Objects.Copy) {
+        authorizationHandler.handleCopyObject(
+            createGCSURI(
+                ((Storage.Objects.Copy) request).getSourceBucket(),
+                ((Storage.Objects.Copy) request).getSourceObject()),
+            createGCSURI(
+                ((Storage.Objects.Copy) request).getDestinationBucket(),
+                ((Storage.Objects.Copy) request).getDestinationObject()));
+      } else if (request instanceof Storage.Objects.Patch) {
+        authorizationHandler.handlePatchObject(
+            createGCSURI(bucketName, ((Storage.Objects.Patch) request).getObject()));
+      }
+
+      // Bucket
+      else if (request instanceof Storage.Buckets.List) {
+        authorizationHandler.handleListBucket();
+      } else if (request instanceof Storage.Buckets.Insert) {
+        authorizationHandler.handleInsertBucket(createGCSURI(bucketName, null));
+      } else if (request instanceof Storage.Buckets.Get) {
+        authorizationHandler.handleGetBucket(createGCSURI(bucketName, null));
+      } else if (request instanceof Storage.Buckets.Delete) {
+        authorizationHandler.handleDeleteBucket(createGCSURI(bucketName, null));
+      }
+  }
```

Comment:
It seems that we need to fail if this is unknown method, otherwise it can become a security hole.